lib/net/ssh/gateway.rb

```diff
@@ -99,28 +99,23 @@ def shutdown!
   #   # ...
   #   gateway.close(port)
   #
+  # This function takes variable arguments.  See comments in the case statement
+  # for details.
+  #
+  # The +local_host+ parameter specifies which network interface to bind to locally,
+  # and defaults to "127.0.0.1" if omitted.
   # If +local_port+ is not specified, the next available port will be used.
-  def open(host, port, local_port=nil)
-    ensure_open!
-
-    actual_local_port = local_port || next_port
-
-    @session_mutex.synchronize do
-      @session.forward.local(actual_local_port, host, port)
-    end
-
-    if block_given?
-      begin
-        yield actual_local_port
-      ensure
-        close(actual_local_port)
-      end
+  def open(*args)
```

Comment:
We need to keep documentation in a single place. Also it probably makes sense to follow the argument options what we have in net-ssh eg local_host in front https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/service/forward.rb and not change that. Or go with keyword arguments.